newdle/api.py

```diff
@@ -323,36 +324,68 @@ def create_newdle(title, duration, timezone, timeslots, participants, private, n
             ),
         },
     )
+    db.session.commit()
     return NewdleSchema().jsonify(newdle)
 
 
-@api.route('/newdle/<code>')
-@allow_anonymous
-def get_newdle(code):
-    newdle = Newdle.query.filter_by(code=code).first_or_404(
-        'Specified newdle does not exist'
-    )
-    if newdle.deleted:
-        return DeletedNewdleSchema().jsonify(newdle)
-    return RestrictedNewdleSchema().jsonify(newdle)
-
-
 @api.route('/newdle/<code>', methods=('PATCH',))
-@use_args(UpdateNewdleSchema(), locations=('json',))
+@use_args(NewNewdleSchema(partial=True), locations=('json',))
 def update_newdle(args, code):
     newdle = Newdle.query.filter_by(code=code).first_or_404(
         'Specified newdle does not exist'
     )
-    if newdle.creator_uid != g.user['uid']:
+    if g.user is None or newdle.creator_uid != g.user['uid']:
         raise Forbidden
+    new_participants = []
+    if 'participants' in args:
+        participants = args.pop('participants')
+        # Filter the new participants to be created (left diff)
+        new_participants = [
+            Participant(**_p)
+            for _p in participants
+            if not any(_p['auth_uid'] == p.auth_uid for p in newdle.participants)
```

Comment:
that nested loop with `p` and `_p` is a bit nasty ;) why not create a set with the existing uids and then check `_p['auth_uid'] in existing_uids`? (also rename `_p` to `p` in that case)

Reply:
Sounds like a good idea. I guess initially I just tried to avoid having two sets (the block below too).